python/port/mpconfigport.h

```diff
@@ -144,3 +144,10 @@ extern const struct _mp_obj_module_t modturtle_module;
     { MP_ROM_QSTR(MP_QSTR_time), MP_ROM_PTR(&modtime_module) }, \
     { MP_ROM_QSTR(MP_QSTR_turtle), MP_ROM_PTR(&modturtle_module) }, \
 
+// Enable setjmp in debug mode. This is to avoid some optimizations done
+// specifically for x86_64 using inline assembly, which makes the debug binary
+// crash with an illegal instruction
+#ifndef NDEBUG
+  #define MICROPY_NLR_SETJMP 1
+#endif
```

Comment:
I'm not sure this is the right way to fix it - but I'm not sure I get where the crash comes from though. The illegal emitted instructions you're talking about are those in `python/src/py/nlrx64.c`? We might want to fix them and PR micropython with the fix?

Reply:
Yes, I did a lot of debugging and the problem is that these instructions do a context switch, and for some reason the debug binary doesn't manage to handle that properly. It might be an upstream issue.